oauth2/mock.go

```diff
@@ -0,0 +1,48 @@
+package oauth2
+
+import (
+	"context"
+
+	"github.com/topfreegames/Will.IAM/models"
+	"github.com/topfreegames/Will.IAM/repositories"
+)
+
+// ProviderBlankMock is a Provider mock will all dummy implementations
+type ProviderBlankMock struct {
```

Comment:
This provider is used only as a mock to some unit/integration tests on Will.IAM, so I've moved it untouched to another file.